core/src/banking_stage.rs

```diff
@@ -540,22 +615,31 @@ impl BankingStage {
                                 working_bank,
                             ));
                         }
-                        new_tx_count += processed;
+
+                        // The difference between all transactions passed to execution and the ones that
+                        // are retryable were the ones that were either:
+                        // 1) Committed into the block
+                        // 2) Dropped without being committed because they had some fatal error (too old,
+                        // duplicate signature, etc.)
+                        //
+                        // Note: This assumes that every packet deserializes into one transaction!
+                        consumed_buffered_packets_count += original_unprocessed_indexes
+                            .len()
+                            .saturating_sub(retryable_transaction_indexes.len());
 
                         // Out of the buffered packets just retried, collect any still unprocessed
                         // transactions in this batch for forwarding
-                        rebuffered_packet_count += new_unprocessed_indexes.len();
+                        rebuffered_packet_count += retryable_transaction_indexes.len();
                         let has_more_unprocessed_transactions =
                             Self::update_buffered_packets_with_new_unprocessed(
                                 original_unprocessed_indexes,
-                                new_unprocessed_indexes,
+                                retryable_transaction_indexes,
                             );
                         if let Some(test_fn) = &test_fn {
                             test_fn();
                         }
                         has_more_unprocessed_transactions
                     } else {
-                        rebuffered_packet_count += original_unprocessed_indexes.len();
```

Comment:
Just checking, was this intentional? Seems better to remove this as you've done because it doesn't really represent the same type of rebuffering done above

Reply:
yeah, exactly. These didn't seem to imply the same "retryable failed transaction" since they weren't even executed, so I removed it